sui/wormhole/sources/governance_message.move

```diff
@@ -31,203 +30,167 @@ module wormhole::governance_message {
-        vaa_hash: Bytes32
-    }
-
-    /// Retrieve governance module name.
-    public fun module_name(self: &GovernanceMessage): Bytes32 {
-        self.module_name
-    }
-
-    /// Retrieve governance action (i.e. payload ID).
-    public fun action(self: &GovernanceMessage): u8 {
-        self.action
+        global: bool
     }
 
-    /// A.K.A. target chain == 0.
-    public fun is_global_action(self: &GovernanceMessage): bool {
-        self.chain == 0
-    }
-
-    /// A.K.A. target chain == `wormhole::state::chain_id()`.
-    public fun is_local_action(self: &GovernanceMessage): bool {
-        self.chain == chain_id()
-    }
-
-    /// Computed keccak256 hash of `VAA` message body.
-    public fun vaa_hash(self: &GovernanceMessage): Bytes32 {
-        self.vaa_hash
+    struct DecreeReceipt {
+        payload: vector<u8>,
+        digest: Bytes32
     }
 
-    /// Destroy `GovernanceMessage` to take governance payload.
-    public fun take_payload(msg: GovernanceMessage): vector<u8> {
-        let GovernanceMessage {
-            module_name: _,
-            action: _,
-            chain: _,
-            vaa_hash: _,
-            payload
-        } = msg;
-
-        payload
+    public fun authorize_verify_global(
+        governance_chain: u16,
+        governance_contract: ExternalAddress,
+        module_name: Bytes32,
+        action: u8
+    ): DecreeTicket {
+        DecreeTicket {
+            governance_chain,
+            governance_contract,
+            module_name,
+            action,
+            global: true
+        }
     }
 
-    /// Passing in a deserialized `VAA`, Wormhole performs additional governance
-    /// checks to validate governance emitter before returning deserialized
-    /// `GovernanceMessage`.
-    ///
-    /// NOTE: It is expected that these VAAs are consumed only once using
-    /// `ConsumedVAAs`. Those contracts that use Guardian governance to perform
-    /// administrative functions are expected to have this container to protect
-    /// against replaying these governance actions.
-    public fun verify_vaa(
-        wormhole_state: &State,
-        verified_vaa: VAA,
-    ): GovernanceMessage {
-        state::check_minimum_requirement<GovernanceMessageControl>(
-            wormhole_state
-        );
-
-        // This VAA must have originated from the governance emitter.
-        assert_governance_emitter(wormhole_state, &verified_vaa);
-
-        // Cache VAA digest.
-        let vaa_hash = vaa::digest(&verified_vaa);
-
-        // Finally deserialize Wormhole payload as governance message.
-        let cur = cursor::new(vaa::take_payload(verified_vaa));
-        let module_name = bytes32::take_bytes(&mut cur);
-        let action = bytes::take_u8(&mut cur);
-        let chain = bytes::take_u16_be(&mut cur);
-        let payload = cursor::take_rest(cur);
-
-        GovernanceMessage { module_name, action, chain, payload, vaa_hash }
+    public fun authorize_verify_local(
```

Comment:
this function needs access control, currently it allows bypassing the governance check entirely I think

Reply:
pushed a fix, @a5-pickle please mark this as resolved if you agree with it